### PR TITLE
Software updates

### PR DESCRIPTION
**Pull request by brianeclow**

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -3,18 +3,42 @@ driver:
...
+  require_chef_omnibus: true
+  network:
+    - [ 'forwarded_port', { guest: 8080, host: 8080 } ]
 
 provisioner:
   name: chef_solo
+  environments_path: "environments"
+  nodes_path: "nodes"
+  data_bags_path: "data_bags"
+  encrypted_data_bag_secret_key_path: "test/data/secret.txt"
```
Comment (christopherdeutsch) [inline .kitchen.yml:18]: this file doesn't exist in git (which is probably good!)

File: providers/cookbook_ci.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  jenkins_job job_name do
+    action :nothing
+    config job_config
+  end
+
+  commands = ["bundle install --deployment"]
+  commands << "bundle exec rake test:foodcritic" if new_resource.foodcritic
+  commands << "bundle exec rake test:chefspec" if new_resource.chefspec
+  commands << "bundle exec berks" if new_resource.serverspec
```
Comment (christopherdeutsch) [inline providers/cookbook_ci.rb:24]: Could the `bundle exec berks` be `bundle exec berks update`? Or perhaps `bundle exec berks verify`? Depending on whether or not we think it is a good idea to change the pins or just to verify they are valid or.. ?

(Or actually a `berks install` might be necessary to make sure that the cookbooks are downloaded. Not sure-- this is new territory for me.)

**Comment by christopherdeutsch** [comment]
Awesome! :+1:

File: .kitchen.yml (hunk 1 above)
Comment (brianeclow) [inline .kitchen.yml:18]: They get created in the next revision when test kitchen is switched to chef zero.

File: providers/cookbook_ci.rb (hunk 2 above)
Comment (brianeclow) [inline providers/cookbook_ci.rb:24]: I have found not all authors check in their lock files for either berks or gem files. At the moment, this confirms the cookbooks are downloaded into the workspace. I was planning on making the commands more customizable based on the user's whim.

**Pull request merged**